### ISSUE TITLE
Dropping Python 2.7 support in 1.2.0

### ISSUE THREAD

**Issue by jleclanche** [jazzband/django-oauth-toolkit#579]
Hi folks, bit of a heads up.

I would like to drop support for Python 2.7 in **DOT 1.2.0** (the major release after the next one, out some time this year hopefully). That means the next release will support 1.11+ only, and the one after 2.0+ only.

Reasoning: Django 2.0 now being out, the latest releases of Django have dropped Python 2.7 support.
Django 1.11 still supports 2.7 and 1.11 will be supported until, roughly, April 2020. I fully intend to continue supporting the 1.1.0 branch with bugfixes until that time.

In other words, I would like to follow in the same footsteps that Django established with 1.11 (->1.1.0) and 2.0 (->1.2.0).

Any immediate reactions?

**Comment by bastbnl** [comment]
lgtm. So you're basically saying you intend to support dot 1.1 just as long as Django 1.11 is supported, right? What about versioning it with 1.11? 

What are your thoughts on features introduced with 1.2.0? Are you backporting?

What about naming dot 1.2.0 dot 2.0.0, since it appears to be tied to the Django 2 release? Are you following the Django python dependencies - so including 3.4?

One more... non technical, but please reconsider adding the plain source again with the new releases (#562 and #532).

**Comment by jleclanche** [comment]
- I don't think it's necessary to make the versions match. 
- Features will not be backported. But this is also why I want to do a 1.1.0 release first that supports both (it's easy and there's already some stuff landed).
- Python support will be followed yes; unless it's too much of a pain to do so, the supported versions of python will match the supported versions of django
- I'm not in charge of releases so i cant say re sdists. DOT wheels aren't binary... python packaging still sucks. :/

**Comment by synasius** [comment]
LGTM :+1: 

> I don't think it's necessary to make the versions match.

yes, I think we should stick to the semantic versioning

> Python support will be followed yes; unless it's too much of a pain to do so, the supported versions of python will match the supported versions of django

That's great! In the past I always checked the support matrix of the DRF project (which usually follows Django)

> I'm not in charge of releases so i cant say re sdists. DOT wheels aren't binary... python packaging still sucks. :/
I added `jazzband` as PyPi maintainer but I can also add you @jleclanche 
What's your username on PyPi?

**Comment by jleclanche** [comment]
My PyPI username is `Adys`.

**Comment by synasius** [comment]
@jleclanche I added you as a maintainer

**Comment by MattBlack85** [comment]
let's do this :heart_eyes:

**Comment by hugovk** [comment]
DOT is already > 50% Python 3 👍 

Here's the pip installs for django-oauth-toolkit from PyPI for last month:

| python_version | percent | download_count |
| -------------- | ------: | -------------: |
| 2.7            |  45.39% |         13,732 |
| 3.6            |  39.29% |         11,888 |
| 3.5            |   9.88% |          2,988 |
| 3.4            |   5.39% |          1,630 |
| 2.6            |   0.03% |              9 |
| 3.3            |   0.02% |              6 |
| 3.7            |   0.01% |              2 |

Source: `pypinfo --start-date -57 --end-date -27 --percent --pip --markdown django-oauth-toolkit pyversion`

**Comment by jleclanche** [comment]
1.2.0 has been released.

**Comment by bastbnl** [comment]
Thanks for 1.2.0 @jleclanche. Just a couple of questions:

I knew the 1.2.0 release was going to require python 3. I thought - and never bothered to ask - python 3 would include python 3.4 and 3.5. pypi has it listed for version 3.6. Is that intentional? I'm asking because I'm on 3.5 and the platform does not offer version 3.6 yet. 

Can I kindly ask you to upload the source again? Please consider or reconsider including this in the release automatically.

**Comment by hugovk** [comment]
PyPI lists 3.4-3.6:

https://pypi.org/project/django-oauth-toolkit/

Presumably from the classifiers in:

https://github.com/jazzband/django-oauth-toolkit/blob/master/setup.cfg


tox also includes 3.4-3.6:

https://github.com/jazzband/django-oauth-toolkit/blob/master/tox.ini

But the README says 3.5+:
https://github.com/jazzband/django-oauth-toolkit/blob/master/README.rst

Travis tests 3.4 and 3.6:

https://github.com/jazzband/django-oauth-toolkit/blob/master/.travis.yml
https://travis-ci.org/jazzband/django-oauth-toolkit

**Comment by jleclanche** [comment]
@bastbnl Yes it should work fine on 3.4 and 3.5. The classifiers are missing -- my bad, but not worth a reupload.

**Comment by bastbnl** [comment]
@jleclanche ok, will give it a try. No problem for not uploading a new package for classifiers. But could you please upload source again?
